### PR TITLE
Enable running some simple proofs on CI with Java backend

### PR DESCRIPTION
**Pull request by ehildenb**
This also enables building the Haskell backend, but does not run the proofs with them on CI.

@hjorthjort if you could confirm that you can run `make build-java ; make test-proof` before approving (along with your regular review process), I would appreciate it.

@vladciobanu if you could confirm that you can do `make deps -B ; make build-haskell ; make TEST_SYMBOLIC_BACKEND=haskell test-proof` (not that the proofs pass, just that it does the right thing), that would be nice too.

### CODE REVIEW

**Comment by hjorthjort** [comment]
If I first make sure my repo and deps are up to date with
`git pull --recurse-submodules; make clean; make deps`

I get a failure:
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for K Framework Tool Parent 1.0-SNAPSHOT:
[INFO] 
[INFO] K Framework Tool Parent ............................ SUCCESS [  0.188 s]
[INFO] K Framework KORE ................................... SUCCESS [  0.846 s]
[INFO] K Framework Tool Kernel ............................ SUCCESS [  1.653 s]
[INFO] K Framework KTree .................................. SUCCESS [  0.021 s]
[INFO] K Framework Ocaml Backend .......................... SUCCESS [  0.024 s]
[INFO] K Framework Java Backend ........................... SUCCESS [  0.273 s]
[INFO] K Framework Haskell Backend ........................ FAILURE [ 27.828 s]
[INFO] K Framework LLVM Backend ........................... SKIPPED
[INFO] K Framework Tool Distribution ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  31.460 s
[INFO] Finished at: 2019-02-16T10:44:22+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.7:run (build-haskell) on project haskell-backend: An Ant BuildException has occured: exec returned: 1
[ERROR] around Ant part ...<exec failonerror="true" dir="/home/hjort/code/wasm/wasm-semantics/.build/k/haskell-backend/src/main/native/haskell-backend" executable="stack">... @ 4:147 in /home/hjort/code/wasm/wasm-semantics/.build/k/haskell-backend/target/antrun/build-main.xml
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :haskell-backend
make: *** [Makefile:33: .build/k/make.timestamp] Error 1
-> [2]
```

**Comment by ehildenb** [comment]
@hjorthjort I'll need the bit before the error message to know what's going on. This symptom is usually a result of either (1) submodules not updated fully, which it looks like you did, or (2) The Haskell Stack tool not being installed.

I'll update the README with new instructions too.

**Comment by ehildenb** [comment]
@hjorthjort I've updated the README to include more thorough installation and running instructions. Can you try again following the new instructions?

Also, @cdetrio if you tried and confirmed that the new instructions worked, I would appreciate it. I don't have the MacOS dependencies explicitly listed (only Ubuntu), but I figure you should have most of them already (except maybe Haskell Stack).

File: README.md [hunk 1]
```diff
@@ -1,59 +1,119 @@
...
+
+### Example usage: `./kwasm` runner script
+
+After building the definition, you can run the definition using `./kwasm`.
+The most up-to-date documentation will always be in `./kwasm help`.
+
+Run the file `tests/simple/arithmetic.wast`:
+
+```sh
+./kwasm run tests/simple/arithmetic.wasm
```
Comment (cdetrio) [inline README.md]: should this be arithmetic.wast?

**Comment by cdetrio** [comment]
```
~/wasm-semantics$ ./kwasm prove tests/proofs/simple-arithmetic-spec.k
#True
~/wasm-semantics$ ./kwasm run   --backend java    tests/simple/arithmetic.wast
<generatedTop>
  <k>
    .
  </k>
  <stack>
    .Stack
  </stack>
  <curFrame>
    <addrs>
      .Map
    </addrs>
    <locals>
      .Map
    </locals>
    <globals>
      .Map
    </globals>
  </curFrame>
  <store>
    <funcs>
      .Map
    </funcs>
  </store>
</generatedTop>
```
I guess I haven't installed the haskell backend (`[Error] Critical: I/O Error while executing (Cannot run program "kore-exec" (in directory "."): error=2, No such file or directory)`).

File: README.md (hunk 1 above)
Comment (ehildenb) [reply README.md]: Fixed.

**Comment by ehildenb** [comment]
Yeah, Haskell backend integration is experimental, more for the devs of the haskell backend to debug their prover on it, not for outside consumption. Thanks for double-checking that it works!

@hjorthjort this is ready for review once you get a chance. I don't think it needs a lot of scrutiny, because it's mostly just changes to the build system, and the CI server seems to be happy with the setup. Just as long as you can follow the instructions in the README and run the proof, that should be enough.

**Comment by hjorthjort** [comment]
@ehildenb Sorry for the delay, seems I didn't get notifications of it, or I missed them in my inbox. Will check right away.

**Comment by hjorthjort** [comment]
@ehildenb Still getting the same issues. I do use Stack almost daily, but the Haskell echosystem for Arch Linux is sometimes a bit quirky. Maven and Ant are a bit opaque to me, so I can't really say what is going wrong.

[Here's the output of me running the commands from the readme, in order](https://pastebin.com/raw/c6mLgkPy). The commands are underlined with `============` to make them easier to locate.

**Pull request merged**